### PR TITLE
Fixes for proxy

### PR DESCRIPTION
**Pull request by freesig**

### CODE REVIEW

File: crates/kitsune_p2p/proxy/src/inner_listen.rs [hunk 1]
```diff
@@ -71,14 +71,20 @@ pub async fn spawn_kitsune_proxy_listener(
...
         let i_s_c = i_s.clone();
         tokio::task::spawn(async move {
-            tokio::time::delay_for(std::time::Duration::from_millis(PROXY_KEEPALIVE_MS)).await;
-
-            if i_s_c.req_proxy(proxy_url.clone()).await.is_err() {
-                // either we failed because the actor is already shutdown
-                // or the remote end rejected us.
-                // if it's the latter - shut down our ghost actor : )
-                let _ = i_s_c.ghost_actor_shutdown().await;
+            loop {
```
Comment (neonphog) [inline crates/kitsune_p2p/proxy/src/inner_listen.rs:74]: hah, woops : ) good catch

File: crates/kitsune_p2p/proxy/src/inner_listen.rs [hunk 2]
```diff
@@ -71,14 +71,20 @@ pub async fn spawn_kitsune_proxy_listener(
...
+                tokio::time::delay_for(std::time::Duration::from_millis(PROXY_KEEPALIVE_MS)).await;
```
Comment (neonphog) [inline crates/kitsune_p2p/proxy/src/inner_listen.rs:75]: let's bump this up to 30 seconds - we may also want a backlog item to delete the proxying record if we get an error trying to hook up a channel.

File: crates/kitsune_p2p/proxy/src/inner_listen.rs [hunk 3]
```diff
@@ -323,18 +330,34 @@ impl InternalHandler for InnerListen {
...
+                tls_srv::spawn_tls_server(
+                    short,
+                    base_url,
+                    self.tls_server_config.clone(),
+                    self.evt_send.clone(),
+                    write,
+                    read,
+                );
+            } else {
+                tracing::warn!("Dropping message for {}", dest_proxy_url.as_full_str());
```
Comment (neonphog) [inline crates/kitsune_p2p/proxy/src/inner_listen.rs:347]: good fix :+1:

File: crates/kitsune_p2p/proxy/src/inner_listen.rs [hunk 4]
```diff
@@ -391,14 +414,15 @@ impl InternalHandler for InnerListen {
...
+    #[tracing::instrument(skip(self))]
     fn handle_register_proxy_to(
         &mut self,
         proxy_url: ProxyUrl,
         base_url: url2::Url2,
     ) -> InternalHandlerResult<()> {
         // expire ProxyTo entries at double the proxy keepalive timeframe.
         let expires_at = std::time::Instant::now()
-            .checked_add(std::time::Duration::from_millis(PROXY_KEEPALIVE_MS * 2))
+            .checked_add(std::time::Duration::from_millis(PROXY_KEEPALIVE_MS * 3))
```
Comment (neonphog) [inline crates/kitsune_p2p/proxy/src/inner_listen.rs:425]: can you adjust the comment above to match?

File: crates/kitsune_p2p/proxy/src/inner_listen.rs (hunk 2 above)
Comment (freesig) [reply crates/kitsune_p2p/proxy/src/inner_listen.rs:75]: https://www51.v1host.com/Holo/story.mvc/Summary?oidToken=Story:17568

File: crates/kitsune_p2p/proxy/src/inner_listen.rs (hunk 4 above)
Comment (freesig) [reply crates/kitsune_p2p/proxy/src/inner_listen.rs:425]: I think it's the same const. Are you happy with 3x? or should I put it back to 2x. I might make that multiplier a const too.

**Review by neonphog — APPROVED** [review]

**Pull request merged**